perceval/backends/core/zulip.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2020 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Authors:
+#     Venu Vardhan Reddy Tekula <venuvardhanreddytekula8@gmail.com>
+#
+
+
+import logging
+
+from grimoirelab_toolkit.uris import urijoin
+
+from ...backend import (Backend,
+                        BackendCommand,
+                        BackendCommandArgumentParser)
+from ...client import HttpClient
+
+logger = logging.getLogger(__name__)
+
+CATEGORY_MESSAGE = "message"
+
+
+class Zulip(Backend):
+    """Zulip Backend."""
+
+    version = '0.1.0'
+
+    CATEGORIES = [CATEGORY_MESSAGE]
+
+    def __init__(self, url, stream, api_token, email, origin, tag=None, archive=None, ssl_verify=True):
+        super().__init__(origin, tag, archive, ssl_verify)
+        origin = url
+
+        super.__init__(origin, tag=tag, archive=archive, ssl_verify=True)
+        self.stream = stream
+        self.api_token = api_token
+        self.email = email
+
+        self.client = None
+
+    def fetch(self, category=CATEGORY_MESSAGE):
+        """Fetch the message from the stream."""
+        kwargs = {}
+        items = super().fetch(category, **kwargs)
+
+        return items
+
+    def fetch_items(self, category, **kwargs):
+        """Fetch the messages."""
```

Comment:
It could make sense to have a specific method to extract the messages retrieved

```
         """Fetch the messages."""
 
         anchor = 'oldest'
-        raw_message = self.client.get_message(anchor)
-        print(raw_message)
+        raw_message = self.client.get_messages(anchor)
+        messages = self.parse_messages(raw_message)
+        for msg in messages:
+            yield msg
+
+    def parse_messages(self, raw_messages):
+        result = json.loads(raw_messages)
+
+        return result.get('messages', None)
```